sound/soc/sof/ipc.c

```diff
@@ -270,8 +271,15 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 	/* add message to transmit list */
 	list_add_tail(&msg->list, &ipc->tx_list);
 
-	/* schedule the message if not busy */
-	if (snd_sof_dsp_is_ready(sdev))
+	/* check if is_ready op is defined */
+	is_dsp_ipc_ready = snd_sof_dsp_is_ready(sdev);
+	if (is_dsp_ipc_ready < 0) {
+		spin_unlock_irq(&sdev->ipc_lock);
+		return -EBUSY;
+	}
```

Comment:
sorry, what does this do?
The is_ready op is marked a mandatory 
int (*is_ready)(struct snd_sof_dev *sof_dev); /* mandatory */

The comment seems wrong?